runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -93,6 +100,17 @@
   // Because of setKey(Object), we really must refresh stateInternals() at each access
   private final StepContext stepContext;
 
+  @Nullable
+  private final SchemaCoder<InputT> schemaCoder;
+
+  @Nullable final SchemaCoder<OutputT> mainOutputSchemaCoder;
```

Comment:
Why is mainOutput special in this case ? Don't we need schema coders for other outputs ?

Reply:
It's used for the default output() method, which always produces to the main output.

Reply:
It's used for the default output() method, which always produces to the main output.